src/fireedge/src/modules/components/Forms/VrTemplate/InstantiateForm/Steps/Networking/NicMenu/NicColumn.js

```diff
@@ -0,0 +1,126 @@
+ * ------------------------------------------------------------------------- */
+
+import PropTypes from 'prop-types'
+import { Component } from 'react'
+import { Box, Button, List, ListItem } from '@mui/material'
+import Legend from '@modules/components/Forms/Legend'
+import NicCard from './NicCard'
+import { VnAPI, SecurityGroupAPI } from '@FeaturesModule'
+import { Tr } from '@modules/components/HOC'
+import { T } from '@ConstantsModule'
+
+/**
+ * Renders a column of NICs with actions to add, remove, and select NICs.
+ *
+ * @param {object} props - The props object.
+ * @param {Array} props.nics - An array of NIC objects.
+ * @param {Function} props.addNic - A function to add a new NIC.
+ * @param {Function} props.removeNic - A function to remove a NIC.
+ * @param {Function} props.selectNic - A function to select a NIC.
+ * @param {number} props.activeNic - The index of the currently active NIC.
+ * @returns {Component} NicColumn.
+ */
+const NicColumn = ({ nics, addNic, removeNic, selectNic, activeNic } = {}) => {
+  const { data: vnets } = VnAPI.useGetVNetworksQuery()
+  const { data: secgroups } = SecurityGroupAPI.useGetSecGroupsQuery()
+
+  return (
+    <Box
+      pt={1}
+      display="flex"
+      flexDirection="column"
+      height="100%"
+      position="relative"
+    >
+      <Legend title={Tr(T.VirtualRouterNICConfigured)} />
+      <Box
+        sx={{
+          display: 'flex',
+          flexDirection: 'column',
+          width: '100%',
+          flexGrow: 1,
+          overflowY: 'auto',
+          maxHeight: 'calc(100% - 2px)',
+        }}
+      >
+        <List
+          disablePadding
+          sx={{
+            width: '100%',
+          }}
+        >
+          {Array.isArray(nics) &&
+            nics?.length > 0 &&
+            nics.map((nic, index) => (
+              <ListItem disableGutters key={`${nic?.id}-${nic?.nicId}`}>
+                <NicCard
+                  info={{
+                    ...nic,
+                    ...(nic?.network_id && nic?.network_id !== ''
+                      ? {
+                          network: vnets?.find(
+                            (vnet) => vnet?.ID === nic?.network_id
+                          )?.NAME,
+                        }
+                      : { network: Tr(T.VirtualRouterNICNetworkName) }),
+
+                    ...(nic?.secgroup !== ''
+                      ? {
+                          secgroup: secgroups?.find(
+                            (secgroup) => secgroup?.ID === nic?.secgroup
+                          )?.NAME,
+                        }
+                      : { secgroup: 'sadfasd' }),
+                  }}
+                  removeNic={removeNic}
+                  selectNic={selectNic}
+                  active={index === activeNic}
+                />
+              </ListItem>
+            ))}
+        </List>
+      </Box>
+      <Button
+        variant="contained"
+        color="primary"
+        onClick={() => addNic()}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using Function.prototype.bind and arrows functions as attributes will negatively impact performance in React. Each time the parent is rendered, the function will be re-created and trigger a render of the component causing excessive renders and more memory use.